modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1547 @@
+                .thenReturn(ENDPOINT_ME);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+
+        when(userManager.createMe(anyObject(), anyObject())).thenReturn(user);
+
+        SCIMResponse outputScimResponse = meResourceManager.create(scimObjectString, userManager,
+                attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(outputScimResponse.getResponseMessage());
+
+        //Assertions
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+        String returnedURI = outputScimResponse.getHeaderParamMap().get(SCIMConstants.LOCATION_HEADER);
+        String expectedURI = ENDPOINT_ME + "/" + obj.getString("id");
+        Assert.assertEquals(returnedURI, expectedURI);
+
+    }
+
+    //InternalErrorException
+    @DataProvider(name = "dataForTestCreateProvidedUserManagerHandlerIsNull")
+    public Object[][] dataToTestCreateProvidedUserManagerHandlerIsNull()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+
+        return new Object[][]{
+                {NEW_USER_SCIM_OBJECT_STRING, "userName", null, user, ResponseCodeConstants.CODE_INTERNAL_ERROR}
+        };
+    }
+
+    //InternalErrorException
+    @Test(dataProvider = "dataForTestCreateProvidedUserManagerHandlerIsNull")
+    public void testCreateProvidedUserManagerHandlerIsNull(String scimObjectString, String attributes,
+                                                           String excludeAttributes, Object objectUser,
+                                                           int expectedScimResponseStatus)
+            throws BadRequestException, NotFoundException, CharonException, ConflictException {
+
+        User user = (User) objectUser;
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(InternalErrorException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new InternalErrorException()));
+        when(userManager.createMe(anyObject(), anyObject())).thenReturn(user);
+
+        SCIMResponse outputScimResponse = meResourceManager.create(scimObjectString,
+                null, attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    //InternalErrorException
+    @DataProvider(name = "dataForTestCreateNewlyCreatedUserResourceIsNull")
+    public Object[][] dataToTestCreateNewlyCreatedUserResourceIsNull() {
```

Comment:
```suggestion
    public Object[][] dataToTestCreatedUserResourceIsNull() {
```

Reply:
Fixed in commit a5699e694bc31369fb4a347d4a0a12177e73775d